backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/bpartner/bpartnercomposite/JsonRetrieverService.java

```diff
@@ -776,6 +815,37 @@ public Optional<JsonResponseContact> getContact(@NonNull final ExternalIdentifie
 				.map(c -> toJson(c, bpartnerComposite.getBpartner().getLanguage()));
 	}
 
+	@NonNull
+	public PaymentTermId getPaymentTermId(@NonNull final ExternalIdentifier externalIdentifier, @NonNull final OrgId orgId)
+	{
+		switch (externalIdentifier.getType())
+		{
+			case METASFRESH_ID:
+				return PaymentTermId.ofRepoId(externalIdentifier.asMetasfreshId().getValue());
+
+			case INTERNAL_NAME:
+			case VALUE:
+				final PaymentTermQuery queryBuilder = PaymentTermQuery.builder()
+						.orgId(orgId)
+						.value(externalIdentifier.getType() == ExternalIdentifier.Type.VALUE ? externalIdentifier.asValue() : null)
+						.name(externalIdentifier.getType() == ExternalIdentifier.Type.INTERNAL_NAME ? externalIdentifier.asInternalName() : null)
```

Comment:
there is no `C_PaymentTerm.InternalName`.  And `C_PaymentTerm.Name` be edited by a user can any time, so i don't think it makes sense to use it.

Reply:
the thing is.. `C_PaymentTerm.Name` is the only column with a unique index... but I guess we should just add another one for `value` + `org`... but it might be that there will be some duplicates on various DBs :-?

Reply:
hi @cp-ps 
> but it might be that there will be some duplicates on various DBs :-?

in that case we need to deal with it., yo please go ahead and add that index.